clients/payment/src/pages/PaymentPage.tsx

```diff
@@ -1,50 +1,481 @@
+      sanitize: value => value.trim(),
+      validate: (value, { getFieldState }) => {
+        const country = getFieldState("country").value;
+        if (sepaNonEeaCountries.includes(country)) {
+          return validateRequired(value);
+        } else {
+          return;
+        }
+      },
+    },
+    postalCode: {
+      initialValue: paymentLink.billingAddress?.postalCode ?? "",
+      sanitize: value => value.trim(),
+      validate: (value, { getFieldState }) => {
+        const country = getFieldState("country").value;
+        if (sepaNonEeaCountries.includes(country)) {
+          return validateRequired(value);
+        } else {
+          return;
+        }
+      },
+    },
+    state: {
+      initialValue: paymentLink.billingAddress?.state ?? "",
+      sanitize: value => value.trim(),
+    },
+  });
+
+  const [addSepaDirectDebitPaymentMandateData, addSepaDirectDebitPaymentMandate] = useUrqlMutation(
+    AddSepaDirectDebitPaymentMandateFromPaymentLinkDocument,
   );
 
-  return (
-    <>
-      {match(data)
-        .with(AsyncData.P.NotAsked, () => null)
-        .with(AsyncData.P.Loading, () => (
-          <LoadingView color={colors.gray[400]} style={styles.base} />
-        ))
-
-        .with(
-          AsyncData.P.Done(Result.P.Ok({ merchantPaymentLink: P.select() })),
-          merchantPaymentLink => {
-            return merchantPaymentLink == null ? (
-              <NotFoundPage />
-            ) : (
-              <PaymentForm paymentLink={merchantPaymentLink} />
-            );
+  const onPressSubmit = () => {
+    submitForm(values => {
+      if (
+        hasDefinedKeys(values, [
+          "paymentMethod",
+          "iban",
+          "country",
+          "name",
+          "addressLine1",
+          "city",
+          "postalCode",
+          "state",
+        ])
+      ) {
+        const { name, addressLine1, addressLine2, city, country, state, postalCode, iban } = values;
+
+        addSepaDirectDebitPaymentMandate({
+          input: {
+            debtor: {
+              IBAN: iban,
+              address: {
+                addressLine1,
+                addressLine2,
+                country,
+                city,
+                postalCode,
+                state,
+              },
+              name,
+            },
+            paymentLinkId: paymentLink.id,
           },
-        )
-        .otherwise(() => (
-          <ErrorView />
-        ))}
-    </>
+        })
+          .mapOk(data => data.addSepaDirectDebitPaymentMandateFromPaymentLink)
+          .mapOkToResult(data => (isNotNullish(data) ? Result.Ok(data) : Result.Error(undefined)))
+          .mapOkToResult(filterRejectionsToResult)
+          .tapOk(data => {
+            setMandateUrl(data.paymentMandate.mandateDocumentUrl);
+            Router.replace("Success", { paymentLinkId: paymentLink.id });
+          })
+          .tapError(error => {
+            // Router.replace("Error");
+            console.log(error);
```

Comment:
small `console.log` kept here, also about the commented bit, are we supposed to route to error or leave the form as it is?